packages/mds-metrics-service/tests/index.spec.ts

```diff
@@ -0,0 +1,261 @@
+          metric.time_bin_start >= start_time &&
+          metric.time_bin_start <= end_time
+      )
+    }
+  })
+
+  testQuery(() => {
+    const [time_bin_size] = TEST_TIME_BIN_SIZES
+    const [timestamp] = TEST_TIMESTAMPS
+    const { start_time, end_time } = timeframe(time_bin_size, timestamp)
+    const [provider_id] = TEST_PROVIDER_IDS
+    return {
+      params: {
+        required: {
+          name: TEST_METRIC_NAME,
+          time_bin_size,
+          start_time: timestamp
+        },
+        optional: { provider_id }
+      },
+      expected: TEST_METRICS.filter(
+        metric =>
+          metric.time_bin_size === time_bin_size &&
+          metric.time_bin_start >= start_time &&
+          metric.time_bin_start <= end_time &&
+          metric.provider_id === provider_id
+      )
+    }
+  })
+
+  testQuery(() => {
+    const [, time_bin_size] = TEST_TIME_BIN_SIZES
+    const [timestamp] = TEST_TIMESTAMPS
+    const { start_time, end_time } = timeframe(time_bin_size, timestamp)
+    const [provider_id] = TEST_PROVIDER_IDS
+    const [geography_id] = TEST_GEOGRAPHY_IDS
+    return {
+      params: {
+        required: {
+          name: TEST_METRIC_NAME,
+          time_bin_size,
+          start_time: timestamp
+        },
+        optional: { provider_id, geography_id }
+      },
+      expected: TEST_METRICS.filter(
+        metric =>
+          metric.time_bin_size === time_bin_size &&
+          metric.time_bin_start >= start_time &&
+          metric.time_bin_start <= end_time &&
+          metric.provider_id === provider_id &&
+          metric.geography_id &&
+          metric.geography_id === geography_id
+      )
+    }
+  })
+
+  testQuery(() => {
+    const [, , time_bin_size] = TEST_TIME_BIN_SIZES
+    const [timestamp] = TEST_TIMESTAMPS
+    const { start_time, end_time } = timeframe(time_bin_size, timestamp)
+    const [provider_id1, provider_id2] = TEST_PROVIDER_IDS
+    const [geography_id1, geography_id2] = TEST_GEOGRAPHY_IDS
+    const [vehicle_type1, vehicle_type2] = TEST_VEHICLE_TYPES
+    return {
+      params: {
+        required: {
+          name: TEST_METRIC_NAME,
+          time_bin_size,
+          start_time: timestamp
+        },
+        optional: {
+          provider_id: [provider_id1, provider_id2],
+          geography_id: [geography_id1, geography_id2],
+          vehicle_type: [vehicle_type1, vehicle_type2]
+        }
+      },
+      expected: TEST_METRICS.filter(
+        metric =>
+          metric.time_bin_size === time_bin_size &&
+          metric.time_bin_start >= start_time &&
+          metric.time_bin_start <= end_time &&
+          [provider_id1, provider_id2].includes(metric.provider_id) &&
+          metric.geography_id &&
+          [geography_id1, geography_id2].includes(metric.geography_id) &&
+          [vehicle_type1, vehicle_type2].includes(metric.vehicle_type)
+      )
+    }
+  })
+
+  testQuery(() => {
+    const [, , time_bin_size] = TEST_TIME_BIN_SIZES
```

Comment:
I think it'd be more readable if you had something like `const time_bin_size = minutes(5)` instead.

Reply:
It’s just using one of the sizes that’s already been defined. It doesn’t really matter what it is, but I get what you’re saying.